### DISCUSSION TITLE
how to set, define highlight groups for certain dotfiles ie. tmux.conf & config.ini for polybar

### DISCUSSION THREAD

**Discussion by ipatch** [nvim-treesitter/nvim-treesitter#4406] [Q&A] [answered]
Hello 👋, been using neovim for quite a while now, but haven't ventured down the customization rabbit hole especially when it comes to updating / defining / overriding highlight groups for certain files or filetypes.

i'm running arch linux, with neovim installed using `yay` and am using a single configuration file, ie. a `init.lua` _link below_. i'm using the `navarasu/onedark.nvim` neovim theme that works with treesitter to syntax highlight files. using this theme i have defined / overridden the default background color for neovim with a custom background color that i have defined _link below_. i also overrode the color, highlight group color for the commentstring or the `@Comment` to a `#8b94e` to be the same grey color that github uses for comments.

i make all the above changes to help me read the _code comments_ for certain filetypes. it works great when i open a file such as my `init.lua` to configure my nvim setup, however if i open my `tmux.conf` file the `@Comment` strings are not defined, i assume that there is no treesitter support for `tmux.conf` files. however both neovim and vim have a `tmux.vim` file to help highlight the `tmux.conf` file. on my local arch box, and there some files located in,

```
/usr/share/nvim/runtime/ftplugin/tmux.vim
/usr/share/nvim/runtime/syntax/tmux.vim
/usr/share/vim/vim90/ftplugin/tmux.vim
/usr/share/vim/vim90/syntax/tmux.vim
```

i'm like 90% sure the problem i'm running into is that the theme i'm using relies on treesitter and treesitter "is not aware" (forgive me for lack of grammar) of the `tmux.conf` type of file i see something like the below when i run the following command in neovim,

```
TSPlaygroundToggle
```

![image](https://user-images.githubusercontent.com/613805/222006701-ef13d001-40c7-4fd9-b7b3-dfc992ab1b31.png)

and if i run the same playground toggle command within my `init.lua` buffer i see the below

![image](https://user-images.githubusercontent.com/613805/222007259-3e30d9fc-1aff-4e82-a886-0f5fa7c3c290.png)

i added images because i think that'll help explain the color difference i'm seeing with different "code comments" depending on the file being opened. and as i stated above outside of defining the custom background and `@Comment` for what i believe to be _commentstrings_ i'm unaware of how i can get treesitter to recognize a file such `tmux.conf` i hope my question is clear, and i know it's not concise, but i'd much prefer to add more information than less in hopes of getting a detailed response, and that it can help someone else who may run into this issue.

my current `init.lua` is below

https://github.com/ipatch/dotfiles/blob/8eaeac85c1903d7432ca1dc24d2190bd93efb767/jobs/macos/Users/mr-fancy/config/nvim/init.lua

**Comment by amaanq** [comment, marked as answer]
Hl under cursor

Syntax
  - tmuxComment links to Comment 
  
  Link `Comment` to the same color as your `@comment`

**Reply by ipatch** [reply]
thanks for your reply, not sure how to link a color like that using either lua or vimscript. either way i ended cobbling together the below fix for my issue.

apparently my nvim setup is using a syntax file to define the highlight groups for my `tmux.conf` so i added override for the color of the `Comment`

https://github.com/ipatch/dotfiles/blob/935e8a30de785d768db497721fc3449d0524f454/config/nvim/after/syntax/tmux.vim
